### PR TITLE
Release/8.0

### PR DESCRIPTION
**Pull request by niepi**

### CODE REVIEW

**Comment by DracoBlue** [comment]
Workaround for amqp:

https://github.com/php-amqp/php-amqp/issues/386#issuecomment-754184017

**Comment by niepi** [comment]
upgrade the following extensions

- amqp is on master now, but this should be change for release
- redis 5.1.1 -> 5.3.2
- xdebug 2.9.0 -> 3.02

File: 8.0/Dockerfile [hunk 1]
```diff
@@ -26,8 +26,20 @@ RUN docker-php-ext-configure gd \
     --with-webp
 RUN docker-php-ext-install gd
 
-RUN pecl install amqp-1.9.4
-RUN docker-php-ext-enable amqp
+# Extract PHP source
+RUN docker-php-source extract
```
Comment (OskarStark) [inline 8.0/Dockerfile:30]: Can we run all the amp stuff in one RUN command and add a comment when to remove this code and what should be used instead then?

**Comment by niepi** [comment]
currently waiting for a imagick upgrade since this is not yet compatible with php8

https://bugs.php.net/bug.php?id=80474

File: 8.0/Dockerfile [hunk 2]
```diff
@@ -58,14 +70,14 @@ RUN apt-get update && apt-get install --no-install-recommends libldap2-dev -y \
...
-RUN pecl install xdebug-2.9.0
+RUN pecl install xdebug-3.0.2
 RUN echo "zend_extension=`php -i | grep ^extension_dir | cut -f 3 -d ' '`/xdebug.so" >> /usr/local/etc/php/conf.d/xdebug.ini
 
 RUN docker-php-ext-install soap
 
 ENV COMPOSER_HOME=/composer
-COPY --from=composer:1.10.17 /usr/bin/composer /usr/local/bin/composer1
-COPY --from=composer:2.0.7 /usr/bin/composer /usr/local/bin/composer2
+COPY --from=composer:1.10.19 /usr/bin/composer /usr/local/bin/composer1
```
Comment (OskarStark) [inline 8.0/Dockerfile:79]: Can we keep the same versions like in the other branches and change it later across all versions?

File: 8.0/Dockerfile [hunk 3]
```diff
@@ -58,14 +70,14 @@ RUN apt-get update && apt-get install --no-install-recommends libldap2-dev -y \
...
+COPY --from=composer:2.0.8 /usr/bin/composer /usr/local/bin/composer2
```
Comment (OskarStark) [inline 8.0/Dockerfile:80]: Same for this version

File: build_alpine_version.sh [hunk 4]
```diff
@@ -21,10 +21,10 @@ fi
...
 cd $FOLDER
 
-cp ../php.ini php.ini
-cp ../start-cron start-cron
-cp ../start-cron-alpine start-cron-alpine
-cp ../composer.sh composer
+cp -f ../php.ini php.ini
+cp -f ../start-cron start-cron
+cp -f ../start-cron-alpine start-cron-alpine
+cp -f ../composer.sh composer
```
Comment (OskarStark) [inline build_alpine_version.sh:27]: Why this change?

File: build_version.sh [hunk 5]
```diff
@@ -21,16 +21,16 @@ fi
 
 cd $FOLDER
 
-cp ../php.ini php.ini
-cp ../start-cron start-cron
-cp ../composer.sh composer
+cp -f ../php.ini php.ini
+cp -f ../start-cron start-cron
+cp -f ../composer.sh composer
```
Comment (OskarStark) [inline build_version.sh:26]: Same question?

File: build_version.sh [hunk 6]
```diff
@@ -21,16 +21,16 @@ fi
...
 
 echo "Pull latest php:${PHP_VERSION}-fpm"
 docker pull "php:${PHP_VERSION}-fpm" >> ../build.log
 
 echo "Building exozet/php-fpm:${PHP_VERSION}"
 
-echo "FROM php:${PHP_VERSION}-fpm" >> version-Dockerfile
+echo "FROM php:${PHP_VERSION}-fpm" > version-Dockerfile
```
Comment (OskarStark) [inline build_version.sh:33]: Why this change?

File: 8.0/Dockerfile (hunk 1 above)
Comment (niepi) [reply 8.0/Dockerfile:30]: this is the suggestion from @DracoBlue, but i think we should also wait for a new release amqp version. i will remove this

File: build_version.sh (hunk 6 above)
Comment (niepi) [reply build_version.sh:33]: aligns this with alpine version and make local rebuild possible, because dockerfile will be create new for each build

File: build_version.sh (hunk 5 above)
Comment (niepi) [reply build_version.sh]: i removed this

File: build_alpine_version.sh (hunk 4 above)
Comment (niepi) [reply build_alpine_version.sh]: i removed this

File: 8.0/Dockerfile (hunk 1 above)
Comment (DracoBlue) [reply 8.0/Dockerfile:30]: @niepi please put it into one RUN command, to avoid having the entire php code in one docker build

File: build_version.sh (hunk 6 above)
Comment (OskarStark) [reply build_version.sh:33]: Should be done for all releases, can you open an issue? Thanks 

Plus an issue for the composer upgrade. Thanks

File: 8.0/Dockerfile (hunk 1 above)
Comment (niepi) [reply 8.0/Dockerfile:30]: we probably will not release this with the master version anyway, so I remove this and wait for amqp release

Comment (OskarStark) [reply 8.0/Dockerfile:30]: I am 👍🏻 for releasing it with the master version, it's better to test it against an unstable version and see if all your other code works well with PHP 8, instead of not be able to use this image, because of an missing extension, WDYT?

**Comment by niepi** [comment]
switch imagick to master

**Comment by OskarStark** [comment]
@niepi I pushed a `release/8.0` branch and changed your target branch 👍

**Comment by niepi** [comment]
@OskarStark @DracoBlue any more todos before a release?

**Comment by OskarStark** [comment]
Can you please do a rebase on release/8.0 branch? I commented some code in the workflow file 🤔

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -1,14 +1,14 @@
 name: CI
 
 on:
-    push:
-        branches:
-            - 'release/**'
+#    push:
+#        branches:
+#            - 'release/**'
```
Comment (OskarStark) [inline .github/workflows/ci.yml:6]: You can uncomment this, afterwards good to go IMO

**Comment by niepi** [comment]
@OskarStark before we release this, let me test builds this with 8.0.1, then we can include this

**Comment by OskarStark** [comment]
Can you add it in this PR?

**Comment by niepi** [comment]
I have added PHP 8.0.1 with the last commit

**Review by OskarStark — APPROVED** [review]

File: 8.0/Dockerfile-alpine [hunk 8]
```diff
@@ -18,9 +18,14 @@ RUN cp `php -i | grep ^extension_dir | cut -f 3 -d ' '`/*.so /tmp/
...
-RUN docker-php-ext-enable amqp
-RUN cp `php -i | grep ^extension_dir | cut -f 3 -d ' '`/*.so /tmp/
+
+# install master version of amqp extension, @TODO upgrade to release version when php8 support is ready
+RUN docker-php-source extract \
+    && mkdir /usr/src/php/ext/amqp \
+    && curl -L https://github.com/php-amqp/php-amqp/archive/master.tar.gz | tar -xzC /usr/src/php/ext/amqp --strip-components=1 \
+    && docker-php-ext-install amqp \
+    && docker-php-source delete \
+    && cp `php -i | grep ^extension_dir | cut -f 3 -d ' '`/*.so /tmp/
```
Comment (DracoBlue) [inline 8.0/Dockerfile-alpine:28]: @niepi  what does this cp do?

File: build_version.sh (hunk 6 above)
Comment (OskarStark) [reply build_version.sh:33]: Shall we revert this?

Comment (niepi) [reply build_version.sh:33]: and fix this then for all branches separately?

**Comment by DracoBlue** [comment]
Merged. Thanks a lot @niepi !
